### PR TITLE
Implement Levels, Goal, Doors

### LINKED ISSUE #14: Create a Goal
Create a goal point where if the player enters with the win condition, will transition the player into another level.

### PR DESCRIPTION
**Pull request by jaekit**
Closes #14 

New Features:
- Game scene that handles Level transitions
- Level scene that handles Player spawning
- Door scene is an interactable that teleports the Player to a position
- Goal scene is an interactable that emits `goal_reached` signal
- Two basic pixelated lights, 32x32 and 64x64 included
- Basic tileset, subject to change
- `levels.tres` is a resource holding all levels in a list

Changes:
- Water Tank is an Interactable, along with Door and Goal
- Roombot updated sprite, collision shapes
- Fixed Roombot's Raycast to not see through walls

### CODE REVIEW

File: scripts/player.gd [hunk 1]
```diff
@@ -54,5 +63,7 @@ func set_coyote_jump_enabled(value: bool) -> void:
 func is_coyote_jump_enabled() -> bool:
 	return !$CoyoteJumpTimer.is_stopped()
 
-func player_die() -> void:
-	queue_free()
+
+func die() -> void:
+	# TODO: transition to die state
```
Comment (reyhan0173) [inline scripts/player.gd:68]: Any idea what the death function call or state would look like? Would have to change the logic in the hitbox.tscn that calls the player's death

File: scripts/roombot/bot_patrol.gd [hunk 2]
```diff
@@ -4,7 +4,7 @@ class_name BotPatrol extends Patrol
 
 
 func process_frame(delta: float) -> EnemyState:
-	if parent.player_detection.is_colliding():
+	if parent.player_detection.get_collider() is Player:
```
Comment (reyhan0173) [inline scripts/roombot/bot_patrol.gd:7]: any specific reason you changed to check if the body is a player? Before, the raycast only interacted with a player object by setting its mask to the players layer, so it functionally did the same thing.

File: scripts/player.gd (hunk 1 above)
Comment (jaekit) [reply scripts/player.gd:68]: just call body.die() from Enemy, Player will handle the rest.
Player will probably transition to DeathState where an animation plays, then signal the Game to transition back to the same level (recreate the same level again)

File: scripts/roombot/bot_patrol.gd (hunk 2 above)
Comment (jaekit) [reply scripts/roombot/bot_patrol.gd:7]: Right now, the Raycast ignores walls so the Roombot sees through them.
The Raycast needs to also collide with walls and then check if the collider was Player specifically

Comment (reyhan0173) [reply scripts/roombot/bot_patrol.gd:7]: makes sense :+1:

File: scripts/player.gd (hunk 1 above)
Comment (reyhan0173) [reply scripts/player.gd:68]: sounds good.

**Review by reyhan0173 — APPROVED** [review]
good to merge

**Pull request merged**